math/src/main/java/com/powsybl/math/timeseries/TimeSeriesTable.java

```diff
@@ -0,0 +1,597 @@
+/**
+ * Copyright (c) 2018, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.math.timeseries;
+
+import com.google.common.base.Stopwatch;
+import gnu.trove.list.array.TIntArrayList;
+import gnu.trove.map.TObjectIntMap;
+import gnu.trove.map.hash.TObjectIntHashMap;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.DoubleBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public final class TimeSeriesTable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TimeSeriesTable.class);
+
+    public class Correlation {
+
+        private final String timeSeriesName1;
+        private final String timeSeriesName2;
+        private final double coefficient;
+
+        public Correlation(String timeSeriesName1, String timeSeriesName2, double coefficient) {
+            this.timeSeriesName1 = Objects.requireNonNull(timeSeriesName1);
+            this.timeSeriesName2 = Objects.requireNonNull(timeSeriesName2);
+            this.coefficient = coefficient;
+        }
+
+        public String getTimeSeriesName1() {
+            return timeSeriesName1;
+        }
+
+        public String getTimeSeriesName2() {
+            return timeSeriesName2;
+        }
+
+        public double getCoefficient() {
+            return coefficient;
+        }
+
+        @Override
+        public String toString() {
+            return "Correlation(timeSeriesName1=" + timeSeriesName1 + ", timeSeriesName2=" + timeSeriesName2 +
+                    ", coefficient=" + coefficient + ")";
+        }
+    }
+
+    private class TimeSeriesNameMap {
+
+        private final List<String> indexToName = new ArrayList<>();
+
+        private final TObjectIntMap<String> nameToIndex = new TObjectIntHashMap<>();
```

Comment:
Create a utility class for this kind of BiMap (already used in CompactStringBuffer).

Reply:
Added BiList class